docs: add dedicated code snippets for docs types section (#950)
FuelLabs/fuels-ts: apps/docs/src/guide/types/address.md

# Address

In Sway, the `Address` type serves as a type-safe wrapper around the primitive `b256` type. The SDK takes a different approach and has its own abstraction for the `Address` type.

## AbstractAddress Class

The SDK defines the `AbstractAddress` class, which provides a set of utility functions for easy manipulation and conversion between address formats.

<<< @/../../../packages/interfaces/src/index.ts#address-1{ts:line-numbers}

## Address Class

> Aside conforming with the interface from `AbstractClass` class, the `Address` class defines one property: `bech32Address`, wich is of the [Bech32](./bech32.md) type.

Reviewer: ```suggestion
Besides conforming to the interface of the AbstractClass class, the Address class also defines one property: `bech32Address`, which is of the [Bech32](./bech32.md) type.
```

> Besides conforming to the interface of the AbstractClass class, the Address class also defines one property: `bech32Address`, which is of the [Bech32](./bech32.md) type.

Reviewer 2: ```suggestion
Besides conforming to the interface of the `AbstractAddress`, the `Address` also defines one property; `bech32Address`, which is of the [Bech32](./bech32.md) type.
```

<<< @/../../../packages/address/src/address.ts#address-2{ts:line-numbers}

## Creating an Address

Thanks to the utility functions provided by the `AbstractAddress` class, there are several ways to create an `Address` instance:

### From a Bech32 Address

To create an `Address` from a `Bech32` address, use the following code snippet:

<<< @/../../docs-snippets/src/guide/types/address.test.ts#address-2{ts:line-numbers}

### From a Public Key

To create an `Address` from a public key, use the following code snippet:

<<< @/../../docs-snippets/src/guide/types/address.test.ts#address-3{ts:line-numbers}

### From a 256-bit Address

To create an `Address` from a 256-bit address, use the following code snippet:

> <<< @/../../docs-snippets/src/guide/types/address.test.ts#address-4{ts:line-numbers}

Reviewer: I feel like using a hardcoded b256 string over a call to the utility function in this code snippet would provide more clarity to the reader here.

```ts 
const b256 = 0xbebd3baab326f895289ecbd4210cf886ce41952316441ae4cac35f00f0e882a6

const address = Address.fromB256(b256);
```

## Utility Functions

> The `Address` class also provides some pratical utility functions:

Reviewer: ```suggestion
The `Address` class also provides some practical utility functions:
```

Typo

> 1. Create a new `Address` from an ambiguous source that may be a Bech32 or B256 address:

Reviewer: This list could be structured better like this:

```suggestion
1. `fromString`: Create a new `Address` from an ambiguous source that may be a Bech32 or B256 address:
```

i.e. let's add the name of the helper before the description in each point.
